polaris/benchmark/_base.py

```diff
@@ -96,7 +95,7 @@ class BenchmarkSpecification(BaseArtifactModel):
 
     # Public attributes
     # Data
-    dataset: Union[Dataset, str, dict[str, Any]]
+    dataset: Union[Union[Dataset, CompetitionDataset], str, dict[str, Any]]
```

Comment:
**Suggestion**
This can be simplified to this (since our minimum is Python 3.10):
```suggestion
    dataset: Dataset | CompetitionDataset | str | dict[str, Any]
```